lua/toggleterm/terminal.lua

```diff
@@ -175,6 +169,9 @@ end
 function Terminal:resize(size)
   if self:is_split() then
     ui.resize_split(self, size)
+    if config.get("persist_size") then
+      ui.persistent[self.direction] = size
```

Comment:
Looking through this locally, I wonder if all the persistence logic can't be encapsulated inside the `ui` module, so no other parts of toggleterm need to really think about persisting things, e.g. can this logic be moved into `resize_split` itself? and other logic that directly checks or manipulates the persistence table.

In this case there are a few places where the size is set, but it depends on exposing persistent and adding values to it which exposes the implementation detail of the table etc. I think this would be better as a setter that is just a wrapper around adding to the table but means in the future it can be changed without needing to touch this file

e.g.
```lua
M.persist_size(self)

function M.persist_size(term, size)
 -- this way persistent can be made private and the implementation changed in the future
 -- without affecting other areas of code
 persistent[term.direction] = size 
end
```

That might be a bit garbled let me know if what I'm saying is clear or not, I can always change it myself after, but I just think having persistence related logic in different modules will make changing its behaviour in the future or refactoring it later harder.

Reply:
That sounds like a good idea.